### PR TITLE
Actually mounts and curls an API

### PR DESCRIPTION
**Pull request by myxoh**
This is a way of doing and end-end test of grape with a sample API and actually performing API calls against it. 

I am open to suggestions.

Tests the issue described in: https://github.com/ruby-grape/grape/issues/1901

### CODE REVIEW

**Comment by dblock** [comment]
I like it! Would make sure those command line errors are handled, I believe `system` doesn't properly error out, so you want to be checking for exit code.

**Comment by myxoh** [comment]
I've got rid of the logs. Not sure whether there was something else you were thinking about. Were you thinking about sig trapping? 
```ruby
  Signal.trap("TERM") do
    shutdown()
  end
```

File: spec/integration/real_api/a_real_api_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+describe 'Mounting a real API' do
+  let!(:server_thread) do
+    Thread.new do
+      `fuser -k 9292/tcp  2>/dev/null`
+      system("rackup spec/integration/real_api/example_api/config.ru 2>/dev/null")
```
Comment (dblock) [inline spec/integration/real_api/a_real_api_spec.rb]: I meant that this call will fail without an exception if, for example, `rackup` cannot run. You need to handle the return from it.

Comment (myxoh) [reply spec/integration/real_api/a_real_api_spec.rb]: :+1:

**Comment by dblock** [comment]
Rebase/fixup?

**Comment by myxoh** [comment]
Right. I need to spend some time getting this right, but it's been a very busy month.
Let me close it for now and get back to it

**Pull request closed without merging**